### PR TITLE
Bytecode

### PR DESCRIPTION
**Pull request by Avuxo**
This is not a 100% implementation of the bytecode engine, but its enough that I can start work on the graphics portion.

Sorry it took so long to get this PR out, its basically been sitting still for a little over a month.

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -1,34 +1,13 @@
+#include <stdlib.h>
 #include <stdio.h>
+#include <stdint.h>
 
-#include "env.h"
-#include "state.h"
-#include "window.h"
+#include "bytecode.h"
 
 int main(int argc, char **argv) {
```
Comment (Avuxo) [inline src/main.c:7]: testing script you're welcome to fiddle with if you want to.

File: src/bytecode.h [hunk 2]
```diff
@@ -0,0 +1,78 @@
+#ifndef _KANSAI_KBC_
+#define _KANSAI_KBC_
+
+#include <stdint.h>
+
+#define MAX_OPCODE 10
+#define DEFAULT_DOT_ENTRIES 100
+#define DEFAULT_OPTIONS 4 /* I find it very unlikely we'll see more than 4. */
```
Comment (hsuch) [inline src/bytecode.h:8]: idk about this, I've seen VN and non-VN games with a larger amount of options (for things like moving location, selecting items, not necessarily dialogue choices)

**Review by hsuch — APPROVED** [review]

File: src/bytecode.h (hunk 2 above)
Comment (Avuxo) [reply src/bytecode.h:8]: except that would be a custom UI, not an option selection screen.

File: src/bytecode.c [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+  01 - end asset set
+  02 - load string into current asset set (continues reading until null-byte)
+  03 - load png file based on filename (continues reading until null-byte)
+  04 - load ogg file based on filename (continues reading until null-byte)
+  
+
+  An example bytecode script:
+  01 11 6D 69 61 2e 70 6E 67 00 10 74 65 73 74 20 73 74 72 69 6E 67 00 10
+  6F 70 74 69 6F 6E 31 00 10 6F 70 74 69 6F 6E 32 00 02 03 21 00 01 00 16
+  10 00 01 00 11 00 00 00 1F 00 02 30 00 03 04 00 00 00 00 00 00 00 00 00
```
Comment (pandaxtc) [inline src/bytecode.c]: The bytecode reference is out of date, and doesn't fully explain the instruction format

**Review by pandaxtc — CHANGES_REQUESTED** [review]

**Review by pandaxtc — APPROVED** [review]

**Pull request merged**